src/misc/lv_array.h

```diff
@@ -38,46 +38,159 @@ typedef struct {
  * GLOBAL PROTOTYPES
  **********************/
 
+/**
+ * Init. an array.
+ * @param array pointer to an `lv_array_t` variable to initialize
+ * @param capacity the initial capacity of the array
+ * @param element_size the size of an element in bytes
+ */
 void lv_array_init(lv_array_t * array, uint32_t capacity, uint32_t element_size);
 
-void lv_array_copy(lv_array_t * target, const lv_array_t * array);
+/**
+ * Resize the array to the given capacity.
+ * @note if the new capacity is smaller than the current size, the array will be truncated.
+ * @param array pointer to an `lv_array_t` variable
+ * @param new_capacity the new capacity of the array
+ */
+void lv_array_resize(lv_array_t * array, uint32_t new_capacity);
+
+/**
+ * Deinit. an array. Free the allocated memory.
+ * @param array pointer to an `lv_array_t` variable to deinitialize
+ */
+void lv_array_deinit(lv_array_t * array);
+
+/**
+ * Return how many elements are stored in the array.
+ * @param array pointer to an `lv_array_t` variable
+ * @return the number of elements stored in the array
+ */
+uint32_t lv_array_size(const lv_array_t * array);
+
+/**
+ * Return the capacity of the array, i.e. how many elements can be stored.
+ * @param array pointer to an `lv_array_t` variable
+ * @return the capacity of the array
+ */
+uint32_t lv_array_capacity(const lv_array_t * array);
+
+/**
+ * Return if the array is empty
+ * @param array pointer to an `lv_array_t` variable
+ * @return true: array is empty; false: array is not empty
+ */
+bool lv_array_is_empty(const lv_array_t * array);
+
+/**
+ * Return if the array is full
+ * @param array pointer to an `lv_array_t` variable
+ * @return true: array is full; false: array is not full
+ */
+bool lv_array_is_full(const lv_array_t * array);
+
+/**
+ * Copy an array to another.
+ * @note this will create a new array with the same capacity and size as the source array.
+ * @param target pointer to an `lv_array_t` variable to copy to
+ * @param source pointer to an `lv_array_t` variable to copy from
+ */
+void lv_array_copy(lv_array_t * target, const lv_array_t * source);
 
+/**
+ * Remove all elements in array.
+ * @param array pointer to an `lv_array_t` variable
+ */
 void lv_array_clear(lv_array_t * array);
 
-void lv_array_resize(lv_array_t * array, uint32_t new_capacity);
+/**
+ * Remove the element at the specified position in the array.
+ * @param array pointer to an `lv_array_t` variable
+ * @param index the index of the element to remove
+ * @return true: success; false: error
+ */
+bool lv_array_pop(lv_array_t * array, uint32_t index);
```

Comment:
In other languages `pop` means to **get and remove** an item from a list, but in this case we just remove the item at the given index.

So I think the function name should reflect that: `lv_array_remove(lv_array_t * array, uint32_t index)`